runner/src/runtime/context.rs

```diff
@@ -0,0 +1,32 @@
+use super::RuntimeFlavor;
+use std::cell::RefCell;
+
+thread_local! {
+    static RT_CONTEXT: RefCell<Option<RuntimeFlavor>> = RefCell::new(None);
+}
+
+pub(crate) fn current() -> Option<RuntimeFlavor> {
+    RT_CONTEXT.with(|ctx| ctx.borrow().clone())
+}
+
+pub(crate) fn enter(flavor: RuntimeFlavor) -> EnterGuard {
+    RT_CONTEXT.with(|ctx| {
+        let old = ctx.borrow_mut().replace(flavor);
```

Comment:
I'm a little confused by this, because if used asynchronously wouldn't this cause race conditions where another thread/task would access the modified environment rather than the intended one?

Reply:
This was also another thing I took from tokio. This is behind a `thread_local` so using it asynchronously shouldn't be an issue. You can also take a look at how tokio uses it [context::enter](https://github.com/tokio-rs/tokio/search?q=context%3A%3Aenter)

There is another case where the runtime is multi-threaded, but the runtime will do the thread spawning directly instead of the developer. During thread spawning we also call `enter` for that specific thread to have this runtime context as well. But the current rt implementation is only single threaded.